processor/batchprocessor/batch_processor_test.go

```diff
@@ -80,6 +80,91 @@ func TestProcessorLifecycle(t *testing.T) {
 	}
 }
 
+func BenchmarkBatchTraceProcessor(b *testing.B) {
+	b.StopTimer()
+	cfg := Config{
+		Timeout:                  100 * time.Millisecond,
+		SendBatchSize:            1000,
+		MetadataCardinalityLimit: defaultMetadataCardinalityLimit,
+	}
+	runTraceProcessorWoMetadataBenchmark(b, cfg)
```

Comment:
why do we call different functions in the benchmarks? the idea is to have only `MetadataKeys:             []string{"X-Sf-Token"},` line different between benchamarks

Reply:
in the second case, I am putting the metadata in the context. i did not want to add a if else, so i separated it

Reply:
We don't need to put metadata in the context. We are benchmarking if it's safe to add `MetadataKeys: "X-Sf-Token"` by default for non-passthrough (typical) workloads